cl/corpus_importer/management/commands/recap_into_opinions.py

```diff
@@ -0,0 +1,124 @@
+
+from cl.lib.command_utils import logger
+from cl.lib.microservice_utils import microservice
+from cl.search.models import (
+    SOURCES,
+    Court,
+    Docket,
+    Opinion,
+    OpinionCluster,
+    RECAPDocument,
+)
+
+
+def import_opinions_from_recap(court=None, total_count=0):
+    """Import recap documents into opinion db
+
+    :param court: Court ID if any
+    :param total_count: The number of new opinions to add
+    :return: None
+    """
+    count = 0
+    if not court:
+        courts = Court.objects.filter(jurisdiction=Court.FEDERAL_DISTRICT)
+    else:
+        courts = Court.objects.filter(id=court)
+    for court in courts:
+        logger.info(f"Importing RECAP documents for {court}")
+        cluster = (
+            OpinionCluster.objects.filter(docket__court=court)
+            .order_by("-date_filed")
+            .first()
+        )
+        if not cluster:
+            print("No clusters for this court")
+            logger.info(f"Court {court} has no opinion clusters")
+
+            continue
+        documents = RECAPDocument.objects.filter(
+            docket_entry__docket__court=court,
+            docket_entry__docket__date_filed__lte=cluster.date_filed,
+            is_available=True,
+            is_free_on_pacer=True,
+        ).order_by("docket_entry__docket__date_filed")
+
+        chunk_size = 100
+        paginator = Paginator(documents, chunk_size)
+        for page_number in paginator.page_range:
+            chunk = paginator.page(page_number).object_list
+            for recap_document in chunk:
+                docket = Docket.objects.get(
+                    docket_entries__recap_documents__id=recap_document.id
+                )
+                if "-cv-" not in docket.docket_number:
+                    logger.info(f"Skipping non civil opinion")
+                    continue
+
+                ops = Opinion.objects.filter(sha1=recap_document.sha1)
+                if ops.count() > 0:
+                    print("Skipping previously imported opinion")
+                    continue
+
+                response = async_to_sync(microservice)(
+                    service="recap-extract",
+                    filepath=recap_document.filepath_local,
+                    params={"strip_margin": True},
+                )
+                with transaction.atomic():
+                    cluster = OpinionCluster.objects.create(
+                        case_name_full=docket.case_name_full,
+                        case_name=docket.case_name,
+                        case_name_short=docket.case_name_short,
+                        docket=docket,
+                        date_filed=recap_document.docket_entry.date_filed,
+                        source=SOURCES.RECAP,
+                    )
+                    Opinion.objects.create(
+                        cluster=cluster,
+                        type=Opinion.TRIAL_COURT,
+                        plain_text=response.json()["content"],
```

Comment:
Do we need a check to make sure doctor worked before creating this object? Something like `response.raise_for_status()`?

Reply:
yeah that makes sense to me